.travis.yml

```diff
@@ -1,12 +1,47 @@
+sudo: false
 language: python
 python: 2.7
+
+# This is a big repository so break Travis CI work into 4 CI jobs.
+env:
+  - CHUNK=0
+  - CHUNK=1
+  - CHUNK=2
+  - CHUNK=3
+
+before_install:
+  - export GALAXY_REPO=https://github.com/galaxyproject/galaxy
+  - export GALAXY_RELEASE=release_16.07
+  - export PLANEMO_CONDA_PREFIX="$HOME/conda"
+
 install:
-  - pip install click==5.1
-  - pip install --upgrade setuptools
-  - pip install planemo
+  - pip install flake8 planemo
+  - planemo conda_init
+  - export PATH="$PLANEMO_CONDA_PREFIX/bin:$PATH"
+  - conda create -y -q -c bioconda --name iuc_conda samtools=0.1.19 bcftools
+  - . activate iuc_conda
+  - planemo --version
+  - conda --version
+  - if ! git diff --quiet "$TRAVIS_COMMIT_RANGE" -- ; then TRAVIS_COMMIT_RANGE=master...; fi
+  - echo $TRAVIS_COMMIT_RANGE
+  - |
+      planemo ci_find_repos --exclude_from .tt_blacklist \
+                            --exclude packages --exclude data_managers \
+                            --changed_in_commit_range "$TRAVIS_COMMIT_RANGE" \
+                            --chunk_count 4 --chunk "${CHUNK}" \
+                            --output changed_repositories_chunk.list
+  - cat changed_repositories_chunk.list
 
 script:
-  - planemo --version
-  - planemo shed_lint --report_level warn --tools --fail_level error --recursive tools/
-  - planemo shed_lint --report_level warn --tools --fail_level error --recursive packages/
-  - planemo shed_lint --report_level warn --tools --fail_level error --recursive suites/
+  - cd "$TRAVIS_BUILD_DIR"
+  - while read -r DIR; do planemo shed_lint --tools --ensure_metadata --urls --report_level warn --fail_level error --recursive "$DIR" || exit 1; done < changed_repositories_chunk.list
+  - while read -r DIR; do planemo conda_install "$DIR"; done < changed_repositories_chunk.list
+  - while read -r DIR; do planemo test --conda_dependency_resolution --galaxy_branch "$GALAXY_RELEASE" --galaxy_source "$GALAXY_REPO" "$DIR" || exit 1; done < changed_repositories_chunk.list
+
+#after_success:
```

Comment:
@drosofff we could also automatically upload the tools to the toolsheds by uncommenting the following lines and setting SHED_PASSWORD and SHED_EMAIL in the travis settings. The IUC is doing it like this. We could maybe try doing this for the testtoolshed and see how it goes?